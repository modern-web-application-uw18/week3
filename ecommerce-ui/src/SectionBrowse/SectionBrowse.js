import React,{Component} from 'react';
import PropTypes from 'prop-types';

import Card from '../Card/Card.js';

export default class SectionBrowse extends Component {

    render() {
      const airbnbs = this.props.airbnbs;
      const places = airbnbs.map((place, index) => {
        return <Card place={place} image={place.image} key={index} index={index} addToCart={this.props.addToCart}/>
      });

        return (
          <section className="places">
          <h2>Available Places</h2>
            {places}
          </section>
        );
    }
}

SectionBrowse.propTypes = {
  airbnbs: PropTypes.array.isRequired,
  addToCart: PropTypes.func.isRequired
}

SectionBrowse.defaultProps = {
  airbnbs: [{
    title: 'test_title',
    houseType: 'Condo',
    image: 'test_image',
    location: {
      city: 'honolulu',
      country: 'USA'
    },
    payment: {
      cost: 999,
      description: 'Free cancellation'
    },
    host: {
      name: 'Eric',
      isSuperhost: true
    },
    rating: {
      stars: 5,
      review: 999
    }
  }],
  addToCart: () => {
    console.log('add to cart test function');
  }
}
