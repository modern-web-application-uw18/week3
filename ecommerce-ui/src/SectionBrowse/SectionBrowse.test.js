import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';

import SectionBrowse from './SectionBrowse.js';

describe('SectionBrowse', () => {
  it('Smoke Test', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SectionBrowse />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Match Screenshot', () => {
    const component = TestRenderer.create(<SectionBrowse />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  
})
