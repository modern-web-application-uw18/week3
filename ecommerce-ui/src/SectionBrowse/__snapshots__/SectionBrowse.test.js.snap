// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionBrowse Match Screenshot 1`] = `
<section
  className="places"
>
  <h2>
    Available Places
  </h2>
  <div
    className="card"
  >
    <div
      className="coverImage"
      style={
        Object {
          "background": "url('test_image') center / cover no-repeat",
        }
      }
    />
    <div
      className="cardInfo"
    >
      <div
        className="houseInfo"
      >
        <div
          className="location"
        >
          Condo
           - 
          USA
        </div>
        <div
          className="title"
        >
          test_title
        </div>
        <div
          className="text"
        >
          $
          999
           
          <span>
            - 
            <span
              className="payDescription"
            >
              Free cancellation
            </span>
          </span>
        </div>
        <div>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 "
            data-icon="star"
            data-prefix="fas"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 "
            data-icon="star"
            data-prefix="fas"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 "
            data-icon="star"
            data-prefix="fas"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 "
            data-icon="star"
            data-prefix="fas"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 "
            data-icon="star"
            data-prefix="fas"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
           - 
           reviews 
           - Superhost
        </div>
        <div
          className="addToCart"
          onClick={undefined}
        >
          + Add to Cart
        </div>
      </div>
    </div>
  </div>
</section>
`;
